tests/object_test.py

```diff
@@ -936,7 +936,7 @@ def test_thing_with_fd():
 @issue281
 def test_dict_with_fd():
     """Serialize a dict with a file descriptor"""
-    fd = open(__file__, 'r')
+    fd = open(__file__)
```

Comment:
I think we should just change this and other instances of ``fd = open(__file__); fd.close()`` to the ``with open(__file__) as fd:`` context manager, as that's recommended nowadays.

Reply:
Good catch. I'll add a commit to get these using context managers and then start the merges shortly.

Reply:
One small semantic change is that previously we were serializing a closed `fd` file descriptor (because we explicitly call `fd.close()` before using it) and now we'd be serializing an open file descriptor if we were to run the test code inside of the `with` block.

To keep the tests identical to their original behavior we can use `fd` after the `with` context block, but I've always considered that a slightly questionable pattern (with a paranoia that Python might decide to limit the scope of `with` context variable to just the context block one day).

Right now the `fd` variable outlives the block and it'll be closed, so the behavior should be identical. I wouldn't be surprised if some static analysis tool were to start warning about using `with` context-created variables after the block in which they were originally scoped, but that isn't the case now so I'm okay with using `fd` that way for now.

On the flip side, it's kinda interesting that all of our tests were testing a closed file descriptor. Things pass fine if we were to move the test code inside of the `with` block, so I'm kinda leaning towards doing that so that the code reads more simply and so that we actually exercise what happens when an open fd is run through jsonpickle.

Once we do that then we won't be testing closed file descriptors, so I'll add a separate test just for that scenario.

Reply:
I ended up making all of the `open(...)` tests look more or less like this in `object_test.py`:

```python
def test_list_with_fd():
    """Serialize a list with an file descriptor"""
    with open(__file__) as fd:
        _test_list_with_fd(fd)
    _test_list_with_fd(fd)  # fd is closed.


def _test_list_with_fd(fd):
    """Serialize a list with an file descriptor"""
    obj = [fd]
    jsonstr = jsonpickle.encode(obj)
    newobj = jsonpickle.decode(jsonstr)
    assert [None] == newobj
```

I'm cool with that.